### PR TITLE
Kevin/parking refactor

### PR DESCRIPTION
**Pull request by evamo0508**
- fixed flipped Aruco tag in RRL & LLR
- fixed left/right parking ref frame inconsistency
- parking validation script

### CODE REVIEW

File: autovalet_gazebo/launch/setup_slam_test.launch [hunk 1]
```diff
@@ -4,14 +4,16 @@
...
 	<arg name="x" default="-18"/>
 	<arg name="y" default="-9"/>
+	<arg name="theta" default= "0"/>
 
   	<include file="$(find autovalet_gazebo)/launch/parking_lot.launch">
   		<arg name="joystick" value="false"/> 
   		<arg name="world_file" value="$(arg world_file)"/>
 		<arg name="x" value="$(arg x)"/>
 		<arg name="y" value="$(arg y)"/>
+		<arg name="yaw" value= "$(arg theta)"/>
```
Comment (evamo0508) [inline autovalet_gazebo/launch/setup_slam_test.launch:15]: why do we need this?

File: autovalet_parking/src/autovalet_parking/Parker.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+        # Error in x and y in cms
+        trans_err = (self.target_pose[:-1] - self.reached_pose[:-1])*100
+        # Error in orientation (yaw) in degrees
+        # make sure the orientation is the same in hardware so that the "270" works as well
+        yaw_err   = np.abs((self.target_pose[2]*180/3.14 - 270) - self.reached_pose[2]*180/3.14) #comparing the aruco tag (rotated by 270deg) with husky's yaw
+        if yaw_err > 300:
+            yaw_err -= 360
+        print "Target pose (x,y,theta): ", self.target_pose[:-1]*100, self.target_pose[2]*180/3.14 - 270
+        print "Reached pose (x,y,theta): ", self.reached_pose[:-1]*100, self.reached_pose[2]*180/3.14
```
Comment (evamo0508) [inline autovalet_parking/src/autovalet_parking/Parker.py]: np.pi?

File: autovalet_gazebo/launch/setup_slam_test.launch (hunk 1 above)
Comment (kob51) [reply autovalet_gazebo/launch/setup_slam_test.launch:15]: this file is old. why are we using this now?

File: autovalet_parking/src/autovalet_parking/Parker.py [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+        print "Translation error (cms): ", trans_err
+        print "Orientation error (deg): ", yaw_err
+        print "Net error (norm of the error)", np.linalg.norm(trans_err)
+    
+    def get_target_pose(self, gt):
+        
+        self.aruco_tag_gt    = gt.pose[-18] # name 'aruco_visual_marker_7::marker' <to-do> find the name! Don't HC
```
Comment (kob51) [inline autovalet_parking/src/autovalet_parking/Parker.py]: did we wanna do the finding of the name here?

File: autovalet_system/scripts/slam.sh [hunk 4]
```diff
@@ -13,13 +13,13 @@ then
...
             # right turns, left park
-            roslaunch autovalet_gazebo setup_slam_test.launch world_file:=state_machine_rturns_lpark.world x:=0 y:=6 1>> $HOME/slam_log.txt 2>&1
+            roslaunch autovalet_gazebo setup_slam_test.launch world_file:=right_right_left.world x:=0 y:=6 1>> $HOME/slam_log.txt 2>&1
         fi
     else
         if [ "$park" = "right" ]
         then
             # left turns, right park
-            roslaunch autovalet_gazebo setup_slam_test.launch world_file:=state_machine_lturns_rpark.world x:=-6 y:=-9 1>> $HOME/slam_log.txt 2>&1
+            roslaunch autovalet_gazebo setup_slam_test.launch world_file:=left_left_right.world x:=-6 y:=-9 1>> $HOME/slam_log.txt 2>&1
```
Comment (kob51) [inline autovalet_system/scripts/slam.sh]: why did we only  change two filenames? should keep it consistent whatever we choose..

File: autovalet_system/src/autovalet_system.py [hunk 5]
```diff
@@ -127,6 +137,9 @@ def init_detector(self, colorInfo_topic, laneCloud_topic, egoLine_topic):
 
         return LD
 
+    # def published(self):
+
+
```
Comment (kob51) [inline autovalet_system/src/autovalet_system.py]: delete

File: autovalet_system/src/autovalet_system.py [hunk 6]
```diff
@@ -146,85 +159,148 @@ def registered_image_callback(self, color_msg, depth_msg):
...
 
             # if the moveBaseListener is waiting for a new goal, that means we've reached the last goal we've
             # sent, so we're done!
-            if self.moveBaseListener.getState() == MoveBaseState.Waiting:
-                self.prev_state = self.current_state
-                self.current_state = State.FINISH
+            # if self.moveBaseListener.getState() == MoveBaseState.Waiting:
+            #     self.prev_state = self.current_state
+            #     self.current_state = State.FINISH
```
Comment (kob51) [inline autovalet_system/src/autovalet_system.py]: delete

File: autovalet_gazebo/launch/setup_slam_test.launch (hunk 1 above)
Comment (SubramanianKrish) [reply autovalet_gazebo/launch/setup_slam_test.launch:15]: this is for intermediate testing.. we are using this everywhere even now actually. Get's called around in the state machine simulation too! It helps spawn the robot in any orientation that we want

File: autovalet_parking/src/autovalet_parking/Parker.py (hunk 2 above)
Comment (SubramanianKrish) [reply autovalet_parking/src/autovalet_parking/Parker.py]: done

File: autovalet_parking/src/autovalet_parking/Parker.py (hunk 3 above)
Comment (SubramanianKrish) [reply autovalet_parking/src/autovalet_parking/Parker.py]: done

Comment (SubramanianKrish) [reply autovalet_parking/src/autovalet_parking/Parker.py:233]: done

File: autovalet_system/src/autovalet_system.py (hunk 5 above)
Comment (SubramanianKrish) [reply autovalet_system/src/autovalet_system.py]: done

File: autovalet_system/src/autovalet_system.py (hunk 6 above)
Comment (SubramanianKrish) [reply autovalet_system/src/autovalet_system.py]: done

**Pull request merged**